app/controllers/admin/api/applications_controller.rb

```diff
@@ -79,7 +79,7 @@ def application
       applications.joins(:service).where("(services.backend_version <> '1' AND cinstances.application_id = ?)", app_id).first!
 
                      else
-      applications.find(params[:application_id] || params[:id])
+      applications.find(params.require(:application_id) || params.require(:id))
```

Comment:
this will break the logic

You can't require one parameter and use another if he is missing.
If you pass the params without `application_id` it will raise an Error, but it should use the `params[:id]` value.

Reply:
applications.find(params.require(:application_id) || params[:id])

Are you expecting this?

Reply:
No, you still break the logic there with the `require`

```suggestion
      id_params = params.permit(%i[application_id id]).to_h
      applications.find(id_params[:application_id] || id_params[:id])
```

Reply:
See my comment in this PR https://github.com/3scale/porta/pull/2778#discussion_r756826572